apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/components/RareSats/useRareSatsModel.tsx

```diff
@@ -0,0 +1,122 @@
+import { mockedRareSats } from "LLD/features/Collectibles/__integration__/mockedRareSats";
+import { mappingKeysWithIconAndName } from "../Icons";
+import { MappingKeys } from "LLD/features/Collectibles/types/Ordinals";
+import { IconProps } from "LLD/features/Collectibles/types/Collection";
+import { RareSat } from "LLD/features/Collectibles/types/Ordinals";
+import {
+  Satributes,
+  Subrange,
+  SatRange,
+  MockedRareSat,
+  Sat,
+  Icons,
+} from "LLD/features/Collectibles/types/RareSats";
+
+type RareSatsProps = {};
+
+const getIconByKey = (key: string, icons: Icons) => {
+  if (mappingKeysWithIconAndName[key as MappingKeys]) {
+    icons[key] = mappingKeysWithIconAndName[key as MappingKeys].icon;
+  }
+};
+
+const processSatType = (
+  type: string,
+  satributes: Satributes,
+  icons: Icons,
+  displayNames: string[],
+  totalCount: number,
+) => {
+  const attribute = satributes[type as MappingKeys];
+  if (attribute && attribute.count) {
+    displayNames.push(type);
+    getIconByKey(type, icons);
+    totalCount = attribute.count;
+  }
+  return { displayNames, totalCount };
+};
+
+const processSatTypes = (satTypes: string[], satributes: Satributes) => {
+  let displayNames: string[] = [];
+  let totalCount = 0;
+  const icons: { [key: string]: ({ size, color, style }: IconProps) => JSX.Element } = {};
+
+  satTypes.forEach(type => {
+    const result = processSatType(type, satributes, icons, displayNames, totalCount);
+    displayNames = result.displayNames;
+    totalCount = result.totalCount;
+  });
+
+  return { displayNames, totalCount, icons };
+};
+
+const processSubrange = (
+  subrange: Subrange,
+  satributes: Satributes,
+  year: string,
+  value: number,
+) => {
+  const { sat_types } = subrange;
+  const { displayNames, totalCount, icons } = processSatTypes(sat_types, satributes);
+
+  const name = displayNames
+    .map(dn => mappingKeysWithIconAndName[dn.toLowerCase().replace(" ", "_") as MappingKeys]?.name)
+    .filter(Boolean)
+    .join(" / ");
+
+  return {
+    count: totalCount.toString() + (totalCount === 1 ? " sat" : " sats"),
+    display_name: displayNames.join(" / "),
+    year,
+    utxo_size: value.toString(),
+    icons,
+    name,
+  };
+};
+
+const processSatRanges = (satRanges: SatRange[], satributes: Satributes) => {
+  return satRanges.flatMap(range => {
+    const { year, value, subranges } = range;
+    return subranges.flatMap(subrange => processSubrange(subrange, satributes, year, value));
+  });
+};
+
+const processRareSat = (sat: Sat) => {
+  const { extra_metadata } = sat;
+  const satributes = extra_metadata.utxo_details.satributes as Satributes;
+  const satRanges = extra_metadata.utxo_details.sat_ranges;
+  return processSatRanges(satRanges, satributes);
+};
+
+const processRareSats = (mockedRareSats: MockedRareSat[]) => {
```

Comment:
```suggestion
const processRareSats = (rareSats: MockedRareSat[]) => {
```
maybe ? because in the near futur it will not be mocked